File: kotlin-semver/karakum/src/jsMain/resources/annotations/annotateJsPlainObject.js
import ts from "typescript"

const jsoInterfaces = [
    "RangeOptions",
    "Options",
    "CoerceOptions"
]

export default (node) => {
    if (
        ts.isInterfaceDeclaration(node)
        && jsoInterfaces.some(name => node.name.text === name)
    ) {
        return `@js.objects.JsPlainObject`
    }

    return null
}

File: kotlin-testing-library-react/karakum/src/jsMain/kotlin/wrappersgenerator/testing/library/react/annotations/annotateJsPlainObject.kt
package wrappersgenerator.testing.library.react.annotations

import arrow.core.raise.nullable
import io.github.sgrishchenko.karakum.extension.AnnotationContext
import typescript.Node
import typescript.isInterfaceDeclaration

fun annotateJsPlainObject(node: Node, context: AnnotationContext) = nullable {
    ensure(isInterfaceDeclaration(node))
    ensure(
        node.name.text == "RenderHookOptions"
    )

    "@kotlinx.js.JsPlainObject"
}
